examples/delaunay.jl

```diff
@@ -0,0 +1,609 @@
+        model[i].pos = new_position(model, i, t)
+    end
+    return model
+end
+
+# The next component to implement is proliferation. This will be a bit complicated since we need to 
+# know how to sample from the Voronoi cell. Let's start by writing the function that allows us to 
+# identify the Voronoi cell to proliferate. Since the proliferation probabilities are $G_i\Delta t$,
+# we can build a vector that gives the cumulative sum of these probabilities; note that the last
+# entry in this vector will be the sum of all the probabilities, and thus the probability that 
+# any proliferation event occurs.
+function proliferation_probability(model, t)
+    Δt = model.dt
+    probs = zeros(nagents(model)) # Technically nagents is not the number of alive agents, but with the way we are handling agents this is correct
+    for i in allids(model)
+        if !DT.has_vertex(model.triangulation, i) || i in model.dead_cells
+            i > 1 && (probs[i] = probs[i-1])
+            continue
+        end
+        Gᵢ = proliferation_rate(model, i, t)
+        if i > 1
+            probs[i] = probs[i-1] + Gᵢ * Δt
+        else
+            probs[i] = Gᵢ * Δt
+        end
+    end
+    return probs
+end
+function select_proliferative_cell(model, probs)
+    E = probs[end]
+    u = rand() * E
+    i = searchsortedlast(probs, u) + 1 # searchsortedlast instead of searchsortedfirst since we skip over some agents in probs
+    return i
+end
+
+# Since we will not be clearing agents from the model, and instead only 
+# marking them as dead, we use `has_vertex` on the triangulation rather than `hasid` on the model itself.
+#
+# Now let's write the function for sampling from a Voronoi cell. First, for sampling from a triangle,
+# we use the algorithm from [this article](https://blogs.sas.com/content/iml/2020/10/19/random-points-in-triangle.html).
+function sample_triangle(tri::Triangulation, T)
+    i, j, k = triangle_vertices(T)
+    p, q, r = get_point(tri, i, j, k)
+    px, py = getxy(p)
+    qx, qy = getxy(q)
+    rx, ry = getxy(r)
+    a = (qx - px, qy - py)
+    b = (rx - px, ry - py)
+    u₁, u₂ = rand(), rand()
+    if u₁ + u₂ > 1
+        u₁, u₂ = 1 - u₁, 1 - u₂
+    end
+    ax, ay = getxy(a)
+    bx, by = getxy(b)
+    wx, wy = u₁ * ax + u₂ * bx, u₁ * ay + u₂ * by
+    return SVector(px + wx, py + wy)
+end
+
+# Next, we need to know how to select a random triangle from a triangulation. This is done by using a weighted 
+# sample according to the triangle areas. In the function below, there is some obvious room for improvement
+# to reduce allocations, but this is not a concern in this example.
+using StatsBase
+function random_triangle(tri::Triangulation)
+    triangles = collect(DT.each_solid_triangle(tri)) # collect since we need to know the ordering of the areas
+    areas = [DT.triangle_area(get_point(tri, triangle_vertices(T)...)...) for T in triangles]
+    weights = Weights(areas)
+    T = sample(triangles, weights)
+    return T
+end
```

Comment:
```suggestion
# Next, we need to know how to select a random triangle from a triangulation. This is done by using a weighted 
# sample according to the triangle areas.
using StreamSampling
function random_triangle(tri::Triangulation)
    triangles = DT.each_solid_triangle(tri)
    area(T) = DT.triangle_area(get_point(tri, triangle_vertices(T)...)...)
    T = itsample(triangles, area)
    return T
end
```

Reply:
This package is already used in Agents, and it should reduce allocations, actually in this case maybe not, because I ran the example and I saw that `random_triangle` is used only a few times, but anyway this could be useful

Reply:
yes, that is the reason, indeed I see some notable time improvements with `@time step!(initialize_cell_model(radius=0.1, ninit=3), 2000);`